### DISCUSSION TITLE
UnattendedWinstall Discussions

### DISCUSSION THREAD

**Discussion by memstechtips** [memstechtips/UnattendedWinstall#22] [Announcements]
## 👋 Welcome to the UnattendedWinstall Discussions Page!
  Use this page to discuss the UnattendedWinstall project:
  * Ask questions you’re wondering about.
  * Share ideas or improvements.
  * Discuss Issues or Bugs you've encountered.

**Comment by xd003** [comment]
I noticed that this project uses tweaks from winutil. I can recommened some more well regarded debloaters which can you have a look at & cherry pick stuffs you feel is worth adding

https://github.com/builtbybel/BloatyNosy
https://github.com/Raphire/Win11Debloat
https://github.com/farag2/Sophia-Script-for-Windows

**Reply by memstechtips** [reply]
Thanks so much for this, will definitely be having a look at those.

**Comment by pocepoke** [comment]
on Windows 11 23H2v2  cannot install windows
"windows setup encontered an internal error while loading or seaching for an unattend answer file."
setup boot using usb by rufus and don’t select any of the checkboxes in “Customize Your Windows Experience”

**Reply by memstechtips** [reply]
Did you make any changes to the filename? it must be `autounattend.xml` exactly. Otherwise I don't know what to say other than download the file again and create a new Installation USB.

**Reply by pocepoke** [reply]
no I didn't change the name, just download it and place it in root first using anyburn and last using rufus (just place it on the usb)
and on Windows 10 too, I forget the version of windows 10 but downloaded it about 6 months ago
if I delete the autounattend.xml file
Windows installation runs normally

I'll try the IoT one later

now I tried downloading the xml file again and it worked thanks

**Reply by memstechtips** [reply]
Great, I'm glad that it works now.

**Reply by Pulvertoast** [reply]
In case others make this mistake. Be sure to download the ZIP file and do not save the XML using the "Save Link As" command. This breaks the XML. Took me a few tries. Because the installation will not find the XML. Maybe a good tip for everyone.

**Comment by matari-dev** [comment]
thank you so much for all your work in creating the `autounattend.xml` file you've shared!   
I've tried it and it works great thank you 🙇🏼 

I've noticed that any program I run, be it vs code or powershell or anything else, they are run as administrator by default. That is to say when I run the program, I am not specifying `run as administrator`, it just does it automatically.

Is there a way to turn that off? I've been rooting around in settings but couldn't find anything. Does an edit to the registry need to be made?

**Reply by memstechtips** [reply]
Thanks for your feedback and kind words.
I'm not aware of anything I directly added to cause that, will have to investigate.

If you find something, let me know.

**Reply by matari-dev** [reply]
I made a quick video showing me creating a new account on a fresh windows install using the autounattend.xml file you shared ► https://youtu.be/LGxB8BAz8W0

After I create a new user account and log into it for the first time, a terminal window pops up, runs custom script and then restarts the computer.
**My question is:**
Is there a log file that gets generated that shows what the script did?
Is there a way for me to get it to generate a log file?

## Additionally
I didn't mention the last time but the one thing I did different was to create a `ei.cfg` file (_see below_) and put that on the usb install stick as well. Without the `ei.cfg` file, the only option I'm offered is `Win 11 home` but including the file gives me all the Win 11 options including Pro.

I don't imagine that would affect how the answer file works but I just wanted to mention just in case.

## Lastly
Could it be related to the `First Time a New User Logs In` portion of the xml file? (_see below_)

### ei.cfg file
```
[EditionID]


[Channel]
Retail

[VL]
0
```

```xml
<RunSynchronousCommand wcm:action="add">
	<Order>8</Order>
		<Description>Registry Entry to Execute the HKCU Registry Entry Script the First Time a New User Logs In</Description>
	<Path>reg.exe add "HKU\DefaultUser\Software\Microsoft\Windows\CurrentVersion\Runonce" /v "currentuser" /t REG_SZ /d "C:\Windows\Setup\Scripts\currentuser.cmd" /f</Path>
</RunSynchronousCommand>
```

**Reply by memstechtips** [reply]
Hey there.

Yes the CMD window that appears runs the currentuser.cmd file, there is no log for it though. I'm not sure how to generate one doing it the runonce key I'm using.

the ei.cfg file is a nice change, but it would be great if I could include it in the xml file itself.

**Reply by matari-dev** [reply]
I figured it out! The reason why administrator level privileges were always on is because the `User Account Control` Settings were set to `Never notify` (_see image below_). Moving the slider to any setting above that will fix that.

Also if you need the `ei.cfg` file because the installation process doesn't show you all the possible Windows 11 options. You only need to specify one parameter in the file (_see below_).

### ei.cfg file
```
[Channel]
Retail
```


![2024-07-02at2346_UAC](https://github.com/memstechtips/UnattendedWinstall/assets/69375081/c9c38010-9a9a-4859-b0a8-ae6a4453eff6)

**Reply by memstechtips** [reply]
Thanks for this! 
The thing about the ei.cfg is that I want to automatically change it inside of the .xml file and not mess with the Windows ISO.

With regards to the UAC, yes, I have removed the UAC commands that I added in the first versions of the answer file. So the latest answer files has the setting set to the defaults, which is what your screenshot shows. 

Thanks though!

**Reply by matari-dev** [reply]
no problem! thank you again for your work on this! 🙇‍♂️

**Reply by skhaz** [reply]
Quick question: Is it possible to enable auto updates without getting any crap back?

**Reply by memstechtips** [reply]
@skhaz By auto updates, do you mean that you don't have to go and manually install the updates? Because the updates are enabled, it will download the updates, but you have to click "install" for it to install. Honestly, I don't want to touch anything with regards to the updates. 😅

**Comment by PluginMike** [comment]
Hey man, would you be opposed in me making a PR so that we can track semantic-release? This will be done through github actions and will only run on the main branch. It creates an automatic changelog aswell, I saw you are allready writing really good commit messages, I.E feat(something): changed something. Then whatever you change you write in the commit and people can see what version they would want to use, we can also track in the changelog known issues etc.

**Reply by memstechtips** [reply]
How exactly would that work (sorry, I'm new to GitHub)

Currently, I've created a development branch where I will test out a few things before I commit it to the main branch. That way I don't have to commit after every change to the file. I will make bunch of changes at once. Understand?

**Reply by PluginMike** [reply]
I am happy to hop on a Discord call and explain it to you. I already have a GitHub workflow ready.

**Reply by memstechtips** [reply]
Thanks for the suggestion and offer, but I'm not interested in it at this time.

**Comment by skhaz** [comment]
Thank you for the project. Do you accept donations?

**Reply by memstechtips** [reply]
It's a pleasure! I do indeed, and would be very thankful for it, but it's not a necessity. If you want to donate you can use PayPal https://paypal.me/memstech?country.x=ZA&locale.x=en_US

Thanks in advance.

**Comment by flossy83** [comment]
It looks like you've "fixed" Windows 10 for me to a point where I might actually be able to use it.   This would be simply tremendous if it's as good as it seems.   Or is to "too good to be true"?   Time will tell... I'm about to embark on the journey of doing a new build with your xml mods.

I'm still using Windows 7 on my regular machine, however cannot anymore as the newer final Windows 7 driver from NVidia has terrible performance in emulators like Dolphin (cannot roll back to older 2020 version driver as the 3060 didn't exist back then), and Firefox needs a bugfix to be compatible with youtube (their devs are currently preparing to release the bugfix, but it WONT be released to the Windows 7 compatible version of Firefox, so Firefox will be unusable for youtube) and Brave WON'T be having any more updates for Windows 7, and some sites like banking are slowly becoming broken with older browser versions, so this is their way of forcing people to move into their new ecosystem.

So it looks like the future is your mod of Windows 10 and maybe second drive for Linux.

Windows 10 will rapidly be forced into unusability probably within 5 years I'd say.   All they need to do is put something that makes web browsers only support Windows 11 and that's how they'll kick everyone off Windows 10.    You're only as strong as your weakest link, and the web browser and hardware drivers are the weak link.    As soon as Intel stops putting out Windows 10 compatible drivers it's game over.   I bought a 3060 as it is the last generation of GPU to get Windows 7 drivers, but the drivers are bad, much worse than the older 2020 drivers for the 1070 in terms of performance.   This is how they control us by forcing things into an unusable condition (I don't need updates, I just need the damn thing to continue working) but it's okay because it only makes me stronger, only motivates people to use modded versions of Windows and inspires them to try Linux.   The future is a world free of Microsoft and Google, believe it and it will come true.

**Reply by memstechtips** [reply]
I'm glad you found some use in the project!

I'd be happy to hear your feedback on my Answer file(s) once you've tested it out.
From your detailed comment, I can tell you have some advanced computer knowledge, so I would also suggest that you take a look at maybe creating your own answer file which might suit your specific needs better and you can do that here: https://schneegans.de/windows/unattend-generator/

However you don't have to, I just thought I'd mention it!

**Reply by flossy83** [reply]
Cheers, I had a look through Schneegans generator the other day.  I had considered doing my own xml but then I saw your IoT-Enterprise xml is pretty much spot on for what I'm after.   

 My main concern is disabling automatic windows updates and especially disabling that secondary automatic driver updates which I believe is a separate setting from the main one.    In other words, you can disable Windows updates, but it will still automatically download new GPU drivers, motherboard drivers etc. and install them without any notification or user confirmation prompt.    I presume your xml disables this too, but if not it's easy to disable under Advanced System Settings  -> Hardware tab -> device installation settings.

**Reply by ekungurov** [reply]
Frankly speaking I don't know wdym. Windows 10 is very usable without tweaks. But with tweaks it can be a bit better (assuming you know what tweaks are doing).

**Reply by flossy83** [reply]
> I'm glad you found some use in the project!
> 
> I'd be happy to hear your feedback on my Answer file(s) once you've tested it out. From your detailed comment, I can tell you have some advanced computer knowledge, so I would also suggest that you take a look at maybe creating your own answer file which might suit your specific needs better and you can do that here: https://schneegans.de/windows/unattend-generator/
> 
> However you don't have to, I just thought I'd mention it!

Sorry for the long delay, I have finally gotten around to building my new desktop PC with Windows 10 Pro (from Microsoft media creation tool) + your IoT-Like autounattend , and it seems to have gone very well and I am impressed and satisfied with the amount of debloat.    

I did do some additional tweaks:

1. Used gpedit.msc to disable Windows Security's real-time scanning as I noticed that was slowing my WD Black SN850X SSD to about 5MB/sec when copying a folder full of small files.   For it to work I also had to turn off "tamper protection" in Windows Security, otherwise real-time scanning just turns itself back on at the next reboot.  Amazing.

2.  Removed the redundant links under "This PC" in file explorer using these regkeys: [1](https://www.howtogeek.com/222057/how-to-remove-the-folders-from-this-pc-on-windows-10/), [2](https://www.howtogeek.com/331361/how-to-remove-the-3d-objects-folder-from-this-pc-on-windows-10/)

3. Delayed automatic windows security updates for 365 days as those are still left active in the autounattend.xml and I don't trust Microsoft not to package some other undesired software along with their security updates.   Don't worry about me, I've been using Windows 7 without any antivirus, UAC disabled, never had any viruses or been "hacked".    My policy has always been to scan anything I download at virustotal.com before running it, that way I don't have to run a real time scanner and suffer the overhead associated with that, although with real-time scanning disabled the performance is to my liking.

Thanks again for taking the time to make the IoT-like autounattend file, I am making great use of it.

**Comment by KiraZenUnderscore** [comment]
I started using this version of Windows 10, and I can say that I absolutely love it. The only problem is, because of the Cortana being uninstalled, I cannot use the search bar. Is there any way I can fix this? Or at least install the Microsoft Store, because I can't even open the microsoft login pop-up.

**Reply by ekungurov** [reply]
The script uninstalls Microsoft.WindowsStore and Microsoft.XboxApp. Try reinstalling it via PowerShell or winget.

**Comment by Wasabi-Kemosabe** [comment]
I noticed that the Notifications Center doesn't work anymore whenever I click on the date and time button on the taskbar.  Is there anyway to add that back in?

**Reply by skhaz** [reply]
If you want to completely remove game bar, run as admin this

```
reg add HKCR\ms-gamebar /f /ve /d URL:ms-gamebar 2>&1 >''

reg add HKCR\ms-gamebar /f /v "URL Protocol" /d " " 2>&1 >''

reg add HKCR\ms-gamebar /f /v "NoOpenWith" /d " " 2>&1 >''

reg add HKCR\ms-gamebar\shell\open\command /f /ve /d "`"$env:SystemRoot\System32\systray.exe`"" 2>&1 >''

reg add HKCR\ms-gamebarservices /f /ve /d URL:ms-gamebarservices 2>&1 >''

reg add HKCR\ms-gamebarservices /f /v "URL Protocol" /d " " 2>&1 >''

reg add HKCR\ms-gamebarservices /f /v "NoOpenWith" /d " " 2>&1 >''

reg add HKCR\ms-gamebarservices\shell\open\command /f /ve /d "`"$env:SystemRoot\System32\systray.exe`"" 2>&1 >''
```

**Reply by ekungurov** [reply]
Read the source file. Line has a comment: ":: Disables Notification Tray and Calendar"

**Reply by pavl1337** [reply]
I added it back by following **Option Two** from this guide:
https://www.elevenforum.com/t/enable-or-disable-notification-center-in-windows-11.6187/

**Comment by tehmessiah75** [comment]
Is there a way during that install that you can automate joining the device to intune?
I have multiple machines to reconfigure as per below. (I know where some of those settings are in the Schneedgans.de generator but would like to know if this is possible).
Have a predefined local admin, joined to a "LOCAL" workgroup, Join to intune using a "generic@domain" account, remove option to log in with other account, login using a specific 365 account without admin, lock the device down so no software can be installed, disable power settings managed by intune, disable lock screen timeout, hide the switch user option, disable windows hello.

**Reply by ekungurov** [reply]
It's not for home users. And frankly speaking we don't know.

**Reply by memstechtips** [reply]
If you can write a powershell script that can accomplish all of that, yeah, it would be possible.

**Reply by tehmessiah75** [reply]
Its for a set of corporate machines that are all using the same generic account (non-admin) to login with.

I had to migrate all the users from a virtual environment to 365.
After moving them I have to clean the machines of old profiles. (Most time consuming part of the whole thing, which is why I would like to do a full wipe and automate this process)
Create a local admin account.
Rename the machines to "CompanyLocation-PC-DM0(1-6)" and join the LOCAL workgroup.
Then use "Set up a work or school account" > "Join this device to Microsoft Entra ID" (To manage it in intune).
It appears after discussion with a colleague that the rest is managed within intune so I dont need to worry about those other options.

**Comment by nasch007** [comment]
Can't seem to get this to install as a new VM in VirtualBox. It will run the installer then hang after rebooting at "Getting Ready". The icon spins but there seems to be nothing at all happening (no CPU or I/O usage on the host, as well as VirtualBox "Session Information" window. Sincerely confused as to why.

Note, the ISO is good because it installs perfect without the unattend.xml; it also installs with a different unattend.xml (I used the Schneedgan configurator you linked to in another comment).

Uptime was over 4hrs 30mins.

Any help would be greatly appreciated!
![Untitled](https://github.com/user-attachments/assets/52a54933-1888-45be-8157-1ae0e6560234)

**Reply by memstechtips** [reply]
Interesting. Just a question, if you try and install it again and you're stuck on the "Getting Ready" screen. If you click into the window and press the Enter key on your keyboard a few times, can you see if it eventually continues? I'm wondering if it isn't a Powershell script that's "stalled".

**Comment by nasch007** [comment]
I tried a bunch of random keyboard hits but that's a good idea.

At what point/how long after "Getting Ready" should I wait before I do that?

I'm using the latest public build of Win 11 Pro if that makes any difference.

I'll try again and report back.

Thanks!

If you declare with your mouth, "Jesus is Lord," and believe in your heart that God raised him from the dead, you will be saved.
Romans 10:9 (NIV)

Sent from Proton Mail Android

-------- Original Message --------

**Reply by nasch007** [reply]
Well, it's working now. Unfortunately I did the cardinal sin of changing multiple things at once. So I'm not sure exactly what did it.

I deleted the VM and started from scratch. I disabled a few settings in the xml (still want my system clock/calendar). I also duplicated the file to unattend.xml; so both auto and unattend.xml in the .iso -- I did this because a quick read the differences online and it seemed to do with USB/DVD media. I also changed the settings of VirtualBox to use the Hardware RTC as well as change the emulation mode to Hyper-V. Again, not sure which of the settings did it but for anyone in my same situation... try it all and see what sticks hahaha
![starting](https://github.com/user-attachments/assets/9203f061-d5e3-4e72-b47e-369d11c7cf51)

**Reply by memstechtips** [reply]
Glad you got it working now! Also, Amen on the Bible verse, brother.

**Comment by Viking8** [comment]
Hi,

I've been trying to piece together if you had a setting in the autounattend file that disabled the ability to drag a file/folder from one Windows Explorer tab to another and have that tab take focus so you can copy/move said files/folders. On a standard Windows 11 install this works, but not when using your IoT-LTSC-Like unattend file. 
Also when hovering over an icon of an opened program in the taskbar, it should pop a huge preview of that program front and center, but it doesn't when using your file.  Is there a way to get those two things functioning again? 

Thx!

**Reply by ekungurov** [reply]
I believe ExtendedUIHoverTime registry does that, however this script doesn't change it. Check if you have ExtendedUIHoverTime in the registry and remove it if you found it. Also I believe you need GPU driver for this feature to work.

**Reply by ekungurov** [reply]
Also try setting HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM -> EnableAeroPeek to 1. For Windows 11 it doesn't affect anything, but it can for Windows 10, I don't remember exactly

**Comment by JacksADirtyGit** [comment]
So, total noob here. I understand the video and the process. I'm sure that I can follow your SUPERB directions and get this done. I have one reservation before I start. I'm working with a Lenovo Legion laptop. The BIOS installs a few things when doing a clean boot, specifically Vantage and Nahimic and a few other Lenovo addons which I really want removed. As far as I have reviewed this workaround will not help me in my ultimate search to DELETE this Lenovo malware from my laptop. Am I mistaken in my assumptions?

**Comment by pitchedwhite** [comment]
Does it matter if I use the Windows 11 installation media instead of Rufus or Ventoy to do the Windows install with the autounattended.xml file? The current OS I am running was done with the standard autounattended file from around 2-3 weeks ago with Windows installation media, and everything now seems to be worker proper, all the registry edits, debloating, etc. I will note that at the end of the installation process right before I came to the windows start screen where it asked me to set up a local account, make password etc, right before that the powershell command prompt had a small error message saying it could not execute a certain command I will post the picture below from July 5th which is when I did the installation.




![IMG_4618](https://github.com/user-attachments/assets/f0700566-5fad-4372-8126-88fb5e3be9c3)

**Comment by feromateus** [comment]
I don't know what i'm doing wrong, but it not working for me, like the autounattended.xml is literally doing nothing, i have followed the instructions on the youtube, but it just do nothing, why this just only happens to me?

**Reply by nasch007** [reply]
Head to chatGPT and ask the difference between unattended and autounattended.xml. I think mine worked because I had both files present. I'm not entirely sure myself of the difference, but I was able to get it successfully up and running in a VirtualBox VM.

**Comment by nasch007** [comment]
So what is the difference between this minimal install, and what I've been hearing about is called "Tiny11" ?

Would you be willing to do a comparison video on them?

**Comment by pitchedwhite** [comment]
What happens if you use the standard autounattended file and the windows 11 24H2 update rolls out? I know it is heavily tailored for AI is it going to affect the autounatteneded file?

**Comment by AminebajjiDEV** [comment]
Thank you for this setup, it has changed my WIN experience and my computer doesn't go crazy anymore. I upgraded from Win 10 to 11 which was once infected by a trojan virus and even after multiple disinfections and resets it never felt the same again. My laptop feels like new and I hope it stays that way!! And with Winget I can download all the apps I need as I am continually customizing the system to my needs.

**Comment by ExcessiveGBH** [comment]
Clicking on this web site is now doing this "Your connection isn't private
Attackers might be trying to steal your information from [schneegans.de](javascript:void(0);) (for example, passwords, messages, or credit cards).
net::ERR_CERT_INVALID "
I'm using Kaspersky.
I have already used this when 23H2 was end of life to install 25H2 (worked flawlessly) and site was not flagged by Kaspersky.
